src/transformers/image_utils.py

```diff
@@ -91,6 +91,45 @@ def is_batched(img):
     return False
 
 
+def make_list_of_images(images, expected_ndims: int = 3) -> List[ImageInput]:
+    """
+    Ensure that the input is a list of images. If the input is a single image, it is converted to a list of length 1.
+    If the input is a batch of images, it is converted to a list of images.
+
+    Args:
+        images (`ImageInput`):
+            Image of images to turn into a list of images.
+        expected_ndims (`int`, *optional*, defaults to 3):
+            Expected number of dimensions for a single input image. If the input image has a different number of
+            dimensions, an error is raised.
+    """
+    if is_batched(images):
+        return images
+
+    # Either the input is a single image, in which case we create a list of length 1
+    if isinstance(images, PIL.Image.Image):
+        # PIL images are never batched
+        return [images]
+
+    if is_valid_image(images):
+        if images.ndim == expected_ndims + 1:
+            # Batch of images
+            images = [image for image in images]
```

Comment:
You could use images.unbind(0) instead as it's more efficient.

Reply:
@AntreasAntoniou - yep, thanks for the suggestion! We just need to add a check on the type, as this function accepts numpy, tensorflow, jax and pytorch tensors.

Reply:
Decided to leave as it for the moment as including means adding a bunch of if/else logic that isn't super pretty for a small speed up ~1µs

```
In [1]: import numpy as np

In [2]: import torch

In [3]: pt_imgs = torch.randint(0, 256, (16, 3, 20, 20))

In [4]: %timeit [img for img in pt_imgs] 
9.44 µs ± 104 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [5]: %timeit pt_imgs.unbind(0)
8.67 µs ± 113 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```